examples/consumefuel/Program.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Wasmtime;
+
+namespace Example
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using var engine = new Engine(new Config()
+                .WithFuelConsumption(true));
+            using var module = Module.FromTextFile(engine, "consumefuel.wat");
+            using var linker = new Linker(engine);
+            using var store = new Store(engine);
+
+            linker.Define(
+                "",
+                "expensive",
+                Function.FromCallback(store, (Caller caller) =>
+                {
+                    var remaining = store.ConsumeFuel(1000UL);
+                    Console.WriteLine($"Called an expensive function which consumed 1000 fuel. {remaining} units of fuel remaining.");
+                }
```

Comment:
My main goal for making the example project was to make sure I'm not calling ConsumeFuel in a weird way here - I suspect this is how users (myself included) are going to want to call the method but lmk if doing it like this can e.g. cause multithreading issues.

Reply:
I don't foresee any problems given the API's inherent multi-threading issues; lots of types are safe to send between threads (i.e. have no thread-affinity) but not to use from multiple threads simultaneously.